fluss-lake/fluss-lake-paimon/src/test/java/org/apache/fluss/lake/paimon/flink/FlinkUnionReadPrimaryKeyTableITCase.java

```diff
@@ -763,6 +767,134 @@ void testUnionReadPrimaryKeyTableFailover(boolean isPartitioned) throws Exceptio
         jobClient.cancel().get();
     }
 
+    @Test
+    void testUnionReadPartitionsExistInPaimonButExpiredInFluss() throws Exception {
+        // first of all, start tiering
+        JobClient jobClient = buildTieringJob(execEnv);
+
+        String tableName = "expired_partition_pkTable";
+        TablePath tablePath = TablePath.of(DEFAULT_DB, tableName);
+        Map<TableBucket, Long> bucketLogEndOffset = new HashMap<>();
+        // create table & write initial data
+        long tableId =
+                preparePKTableFullType(tablePath, DEFAULT_BUCKET_NUM, true, bucketLogEndOffset);
```

Comment:
I think we can just use `prepareSimplePKTable` to simiplify the test instead of all datatypes.